internal/api/graphql/access/token_auth_method.go

```diff
@@ -0,0 +1,109 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package access
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/golang-jwt/jwt/v5"
+
+	"github.com/cloudoperators/heureka/internal/util"
+)
+
+type ginContextKeyType string
+
+const (
+	ginContextKey   ginContextKeyType = "GinContextKey"
+	scannerNameKey  string            = "scannername"
+	tokenAuthHeader string            = "X-Service-Authorization"
+)
+
+func NewTokenAuthMethod(l Logger, cfg *util.Config) *TokenAuthMethod {
+	if cfg.AuthTokenSecret != "" {
+		return &TokenAuthMethod{logger: l, secret: []byte(cfg.AuthTokenSecret)}
+	}
+	return nil
+}
+
+type TokenClaims struct {
+	Version string `json:"version"`
+	jwt.RegisteredClaims
+}
+
+type TokenAuthMethod struct {
+	logger Logger
+	secret []byte
+}
+
```

Comment:
A  Docstring would probably be good to say: 
- Verifies token presence in header
- Validates token by parsing
- Checks token validity (signature)
- Verifies expiration time

Due to implementation constraints, we intentionally do: 
- No validation of the token issuer (iss claim)
- No validation of the "not before" time (nbf claim)
- No audience validation (aud claim)
- No token signing algorithm verification before parsing

Reply:
I am not a big fan of commenting the code ("good code documents itself" :) ). Would it be ok if I enclose each check in single responsibility functions like:
checkHeaderInRequest
verifyTokenStructure
checkTokenSignature
verifyTokenExpiration
etc.

Reply:
@drochow please check the last commit. I have added some functions to make checks/gets more readable.